apis/logging/v1/output_types.go

```diff
@@ -282,3 +285,28 @@ type Splunk struct {
 	// +optional
 	Fields []string `json:"fields,omitempty"`
 }
+
+// Http provided configuration for sending json encoded logs to a generic http endpoint.
+type Http struct {
+	// Headers specify optional headers to be sent with the request
+	// +optional
+	Headers map[string]string `json:"headers,omitempty"`
+
+	// Timeout specifies the Http request timeout in seconds. If not set, 10secs is used.
+	// +optional
+	Timeout string `json:"timeout,omitempty"`
+
+	// Method specifies the Http method to be used for sending logs. If not set, 'POST' is used.
+	// +kubebuilder:validation:Enum:=GET;HEAD;POST;PUT;DELETE;OPTIONS;TRACE;PATCH
+	// +optional
+	Method string `json:"method,omitempty"`
+
+	// MaxBatch specifies maximum size of batch in bytes
+	// +optional
+	MaxBatch string `json:"maxBatch,omitempty"`
+
+	// Compression specifies the compression type to be used. If not set, defaults to 'none' which does not compress
+	// +kubebuilder:validation:Enum:=gzip;none
+	// +optional
+	Compression string `json:"compression,omitempty"`
```

Comment:
same question here

Reply:
did not find any parameter for compression, have raised a query in fluentd forum

Reply:
https://github.com/fluent/fluentd/discussions/4008

Reply:
/hold

We need to either:
* remove  this from the API since it is not supported by both
* agree to qualifying how it applies to only one collector cc @alanconway